### ISSUE TITLE
[UX] Design review/update for project browser

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1826]
This is a follow up to https://github.com/backdrop/backdrop-issues/issues/1399

Here's my bullet-list of things we need to design/clean-up on the user interface. The project browser interface should be considered as whole. Any designer working on this interface should take this list as a rough guide ONLY and use your own designy-sence where appropriate :)
- [ ] Number of results `Showing 1 to 20 of 127.` 
- [ ] Sort by tabs/links/facets 
- [x] Filter search box (probably should not live in sidebar? see below)
- [ ] Project info in listing:
  - [x] Add to installation queue (look like a button?)
  - [ ] details link (more obvious?)
- [x] Project details pop-up (looking great!!!)
  - [x] Add the 'Add to installation queue' button here too
  - [x] Clicking on project name in modal does nothing. (Remove link? Replace with add to queue link?)
- [ ] Empty text `No results found.` (add `notice` color background?)
- [x] `Next` links on installer should look like buttons

UX: I found myself spending time looking for the search box. I was not expecting it to be in the sidebar because it's not on any other interface. Let's move it to the top of the page and place the cursor in it, just like we do on the modules page.

UX: I expected that clicking on the project name would add it to my installation queue...  Instead the 'deails' modal popped up. This is probably fine as long as we add the 'Add to installation queue' button here too.

UX: I was surprised that I was not taken to the next step after setting my theme as default (since that was the only button on the page). If we make the `Next: Install downloaded modules` link look like a button, this problem will likely be mitigated.

**Comment by docwilmot** [comment]
> Add the 'Add to installation queue' button here too

This was the plan, I tried, failed. Cant figure this one out, sorry. Updating the underlying form from the modal just wont work for me.

Are we getting a designer to work on this though? I'm not one of those.

**Comment by docwilmot** [comment]
All UX suggestions, including the notes on the PR, are done, except the `Add the 'Add to installation queue' button here too`.

Changes list in first post to checklist.

**Comment by jenlampton** [comment]
> Are we getting a designer to work on this though? I'm not one of those.

Yes, but that doesn't need to happen before we merge project browser. Just keeping this issue so we don't forget :)

**Comment by klonos** [comment]
> ...but that doesn't need to happen before we merge project browser.

@jenlampton :+1: this can be a follow-up to polish things.

**Comment by docwilmot** [comment]
> Add the 'Add to installation queue' button here too
This was the plan, I tried, failed. Cant figure this one out, sorry. Updating the underlying form from the modal just wont work for me.

I hate to be wrong, but I tried and didnt fail :smile:  Got this working the hard way by using a button on the dialog, instead of a link and using layout module's `layout-link-button` CSS/JS trick to make it a link. That said, @jenlampton thought it may look better as a button, so just removing the `$form['#attached']` lines where the CSS/JS is included will bring back the button

**Comment by Bojhan** [comment]
Interesting, I did a design for this ages ago in  https://www.flickr.com/photos/28899893@N05/albums/72157623082794892

Some ideas for this implementation: 

- Make the "add link" in the detail modal look more like a button
- Have a max-width on the description, the legibility is a bit poor when it stretches till below the link "add to install queue"

**Comment by docwilmot** [comment]
I think a few followups that are not design reviews may crop up; should we start a new issue/META or repurpose this one?
- add thumbnails for themes and layouts and even modules
- as well as full width previews for themes (maybe with carousel scrolling, like app store previews)
- re-add categories
- re-add author sort
- etc?

**Comment by jenlampton** [comment]
We can repurpose this one :)

- -
Please excuse brevity and typos. Message typed with my thumbs from an Open
Source OS.
